lib/src/components/NxTile/__tests__/NxTile.test.tsx

```diff
@@ -4,68 +4,100 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React from 'react';
-import { shallow } from 'enzyme';
-import 'jest-enzyme';
+import React, { ComponentType } from 'react';
+import { render, screen } from '@testing-library/react';
+import { rtlRenderElement, runTimers, userEvent } from '../../../__testutils__/rtlUtils';
 
 import NxTile from '../NxTile';
 
+function test(Component: ComponentType<{}>, tagName: string, ...classNames: string[]) {
+  return () => {
+    const el = render(<Component/>).container.firstElementChild!;
+    expect(el.tagName).toBe(tagName.toUpperCase());
+
+    for (const className of classNames) {
+      expect(el).toHaveClass(className);
+    }
+  };
+}
+
 describe('NxTile', function() {
-  it('makes a <section> tag with an nx-tile class', function() {
-    expect(shallow(<NxTile/>)).toMatchSelector('section.nx-tile');
-  });
+  it('makes a <section> tag with an nx-tile class', test(NxTile, 'section', 'nx-tile'));
 });
 
 describe('NxTile.Header', function() {
-  it('makes a <header> tag with an nx-tile-header class', function() {
-    expect(shallow(<NxTile.Header/>)).toMatchSelector('header.nx-tile-header');
-  });
+  it('makes a <header> tag with an nx-tile-header class', test(NxTile.Header, 'header', 'nx-tile-header'));
 });
 
 describe('NxTile.HeaderTitle', function() {
-  it('makes a <div> tag with an nx-tile-header__title class', function() {
-    expect(shallow(<NxTile.HeaderTitle/>).children()).toMatchSelector('div.nx-tile-header__title');
+  const renderEl = rtlRenderElement(NxTile.HeaderTitle, {});
+
+  it('makes a <div> tag with an nx-tile-header__title class', test(NxTile.HeaderTitle, 'div', 'nx-tile-header__title'));
+
+  it('renders a tooltip when the title text is too long to be fully seen', async function() {
```

Comment:
I added in this test because the docs specifically mention that a tooltip will be applied when the title text is too long, which is unlike `NxTile.HeaderSubtitle` which just wraps the text. I can remove if it seems unncessary

Reply:
Nice catch! I think it's neat that you spy on the getBoundingClientRect to trigger the overflow. I wonder though if that's testing too much into the implementation details and it might be safer to add a functional test for this. Let me know what you think!

Reply:
> Nice catch! I think it's neat that you spy on the getBoundingClientRect to trigger the overflow. I wonder though if that's testing too much into the implementation details and it might be safer to add a functional test for this. Additionally that will be more future proof because we would be testing the behaviour and if the implementation detail changes we don't have to rewrite this test 😸. Let me know what you think!

Good point. And having looked at the functional tests, there IS a test for the tooltip to appear already. So test has been removed: [de395c4](https://github.com/sonatype/sonatype-react-shared-components/pull/959/commits/de395c41b0125999f64b00b5e78f0008671ec14e)